File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mnStockPrice=prices[0];
        int mxProfit=0;
        for(auto it:prices){
            if(mnStockPrice>it){
                mnStockPrice=it;
            }
            
            mxProfit=max(mxProfit,it-mnStockPrice);
        
        }
        return mxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int len=prices.length;

        int mini=prices[0];
        int profit=0;

        for(int i=0;i<len;i++){
            profit=Math.max(profit,prices[i]-mini);
            mini=Math.min(mini,prices[i]);
        }
        return profit;
        
    }
}